src/core/evaluator.js

```diff
@@ -322,20 +323,20 @@ class PartialEvaluator {
           continue;
         }
         if (xObject.dict.objId) {
-          processed[xObject.dict.objId] = true;
+          processed.put(xObject.dict.objId);
         }
         var xResources = xObject.dict.get("Resources");
         if (!(xResources instanceof Dict)) {
           continue;
         }
         // Checking objId to detect an infinite loop.
-        if (xResources.objId && processed[xResources.objId]) {
+        if (xResources.objId && processed.has(xResources)) {
```

Comment:
I think this should be `processed.has(xResources.objId)`.

Reply:
Of course it should, I really don't know what happened there; thank you!